File: smallpict-php/src/Exceptions/QuotaExceededException.php
<?php

declare(strict_types=1);

namespace SmallPict\Exceptions;

class QuotaExceededException extends SmallPictException
{
    /**
     * @param string $message
     * @param string|null $requestId
     * @param array<string, mixed> $details
     * @param \Throwable|null $previous
     */
    public function __construct(
        string $message = 'Storage or processing quota exceeded',
        ?string $requestId = null,
        array $details = [],
        ?\Throwable $previous = null
    ) {
        parent::__construct($message, 'QUOTA_EXCEEDED', 402, $requestId, $details, $previous);
    }
}

File: smallpict-java/src/main/java/com/smallpict/errors/QuotaExceededException.java
package com.smallpict.errors;

import java.util.Map;

public class QuotaExceededException extends SmallPictException {
    public QuotaExceededException(String message) {
        this(message, null, null);
    }

    public QuotaExceededException(String message, String requestId, Map<String, Object> details) {
        super(message, "QUOTA_EXCEEDED", 402, requestId, details, null);
    }
}
